src/flask_track_usage/storage/couchdb.py

```diff
@@ -0,0 +1,140 @@
+        Executed on "function call".
+
+        :Parameters:
+           - `data`: Data to store.
+        """
+        user_agent = data['user_agent']
+        utcdatetime = datetime.fromtimestamp(data['date'])
+        usage_data = UsageData(url=data['url'],
+                               ua_browser=user_agent.browser,
+                               ua_language=user_agent.language,
+                               ua_platform=user_agent.platform,
+                               ua_version=user_agent.version,
+                               blueprint=data["blueprint"],
+                               view_args=json.dumps(data["view_args"],
+                                                    ensure_ascii=False),
+                               status=data["status"],
+                               remote_addr=data["remote_addr"],
+                               authorization=data["authorization"],
+                               ip_info=data["ip_info"],
+                               path=data["path"],
+                               speed=data["speed"],
+                               datetime=utcdatetime)
+        usage_data.store(self.db)
+
+    def _get_usage(self, start_date=None, end_date=None, limit=500, page=1):
+        """
+        Implements the simple usage information by criteria in a standard form.
+
+        :Parameters:
+           - `start_date`: datetime.datetime representation of starting date
+           - `end_date`: datetime.datetime representation of ending date
+           - `limit`: The max amount of results to return
+           - `page`: Result page number limited by `limit` number in a page
+        """
+        UsageData.by_date.sync(self.db)
+        data = self.db.query(UsageData.by_date.map_fun,
+                             startkey=str(start_date), endkey=str(end_date),
+                             limit=limit)
+        return [row.value for row in data]
+
+
+class CouchDBStorage(_CouchDBStorage):
+    """
+    Creates it's own connection for storage.
+    """
+
+    def set_up(self, database, host='127.0.0.1', port=5984,
+               protocol='http', username=None, password=None):
+        """
+        Sets the collection.
+
+        :Parameters:
+           - `database`: Name of the database to use.
+           - `collection`: Name of the collection to use.
+           - `host`: Host to conenct to. Default: 127.0.0.1
+           - `port`: Port to connect to. Default: 27017
+           - `username`: Optional username to authenticate with.
+           - `password`: Optional password to authenticate with.
+        """
+        import couchdb
+        self.connection = couchdb.Server("{0}://{1}:{2}".format(protocol,
+                                                                host, port))
+        try:
+            self.db = self.connection.create(database)
+        except PreconditionFailed as e:
+            self.db = self.connection[database]
+            print e
+
+
+class UsageData(Document):
```

Comment:
nit: It would be nice to have a docstring here ... but I won't block on merging over it.